networking.go

```diff
@@ -133,6 +136,48 @@ func startStatsdTCP(s *Server, addr *net.TCPAddr, packetPool *sync.Pool) net.Add
 	return listener.Addr()
 }
 
+// startStatsdUnix starts listening for datagram statsd metric packets
+// on a UNIX domain socket address. It does so until the
+// server's shutdown socket is closed. startStatsdUnix returns a channel
+// that is closed once the listening connection has terminated.
+func startStatsdUnix(s *Server, addr *net.UnixAddr, packetPool *sync.Pool) (<-chan struct{}, net.Addr) {
+	done := make(chan struct{})
+	if addr.Network() != "unixgram" {
```

Comment:
I think you can skip this check, it's already handled one layer up.

Reply:
Was following the error checking format similar to `startSSFUnix`, the up level checks only if the address is UnixAddr, and this checks that it's only "unixgram" (ie; not "unix" or "unixpacket"). Happy to remove if you strongly feel so.

Reply:
Yeah, I am pretty +1 to not handling errors in more than one place, especially since it's handled just one function call up too (: